xrpl4j-client/src/main/java/org/xrpl/xrpl4j/client/XrplClient.java

```diff
@@ -252,6 +260,180 @@ public FeeResult fee() throws JsonRpcClientErrorException {
+   * @param sequence The sequence number of the account submitting the {@link Transaction}. A {@link Transaction}
+   *        is only valid if the Sequence number is exactly 1 greater than the previous transaction
+   *        from the same account.
+   * @param account The unique {@link Address} of the account that initiated this transaction.
+   * @return {@code true} if the {@link Transaction} is final/validated else {@code false}.
+   */
+  public Finality isFinal(
+    Hash256 transactionHash,
+    LedgerIndex submittedOnLedgerIndex,
+    UnsignedInteger lastLedgerSequence,
+    UnsignedInteger sequence,
+    Address account
+  ) {
+    return getValidatedTransaction(transactionHash)
+      .map(transactionResult -> {
+        // Note from https://xrpl.org/transaction-metadata.html#transaction-metadata:
+        // "Any transaction that gets included in a ledger has metadata, regardless of whether it is
+        // successful." However, we handle missing metadata as a failure, just in case rippled doesn't perfectly
+        // conform
+        final boolean isTesSuccess = transactionResult.metadata()
+          .map(TransactionMetadata::transactionResult)
+          .filter("tesSUCCESS"::equals)
+          .map($ -> true)
+          .isPresent();
+
+        final boolean metadataExists = transactionResult.metadata().isPresent();
+
+        if (isTesSuccess) {
+          LOGGER.debug("Transaction with hash: {} was validated with success", transactionHash);
+          return Finality.builder()
+            .finalityStatus(FinalityStatus.VALIDATED_SUCCESS)
+            .resultCode(transactionResult.metadata().get().transactionResult())
+            .build();
+        } else if (!metadataExists) {
+          return Finality.builder().finalityStatus(FinalityStatus.VALIDATED_UNKNOWN).build();
+        } else {
+          LOGGER.debug("Transaction with hash: {} was validated with failure", transactionHash);
+          return Finality.builder()
+            .finalityStatus(FinalityStatus.VALIDATED_FAILURE)
+            .resultCode(transactionResult.metadata().get().transactionResult())
+            .build();
+        }
+      }).orElseGet(() -> {
+        try {
+          final boolean isTransactionExpired = FluentCompareTo.is(getMostRecentlyValidatedLedgerIndex())
+            .greaterThan(lastLedgerSequence);
+          if (!isTransactionExpired) {
+            LOGGER.debug("Transaction with hash: {} has not expired yet, check again", transactionHash);
+            return Finality.builder().finalityStatus(FinalityStatus.NOT_FINAL).build();
+          } else {
+            boolean isMissingLedgers = ledgerGapsExistBetween(UnsignedLong.valueOf(submittedOnLedgerIndex.toString()),
+              UnsignedLong.valueOf(lastLedgerSequence.toString()));
+            if (isMissingLedgers) {
+              LOGGER.debug("Transaction with hash: {} has expired and rippled is missing some to confirm if it" +
```

Comment:
This scenario is not covered by the unit tests covering this method (i.e., `isFinal`)